Epochs metadata FutureWarning

This line https://github.com/mne-tools/mne-python/blob/4750f0dd5dc81c15704230c785541ab09fa5373b/mne/epochs.py#L2984

is yielding the traceback:

```
mne\tests\test_epochs.py:4105: in test_make_metadata_bounded_by_row_events
    metadata, events_new, event_id_new = mne.epochs.make_metadata(
mne\epochs.py:2984: in make_metadata
    metadata.iloc[:, 0] = ""
C:\hostedtoolcache\windows\Python\3.11.7\x64\Lib\site-packages\pandas\core\indexing.py:912: in __setitem__
    iloc._setitem_with_indexer(indexer, value, self.name)
C:\hostedtoolcache\windows\Python\3.11.7\x64\Lib\site-packages\pandas\core\indexing.py:1946: in _setitem_with_indexer
    self._setitem_with_indexer_split_path(indexer, value, name)
C:\hostedtoolcache\windows\Python\3.11.7\x64\Lib\site-packages\pandas\core\indexing.py:2039: in _setitem_with_indexer_split_path
    self._setitem_single_column(loc, value, pi)
C:\hostedtoolcache\windows\Python\3.11.7\x64\Lib\site-packages\pandas\core\indexing.py:2158: in _setitem_single_column
    warnings.warn(
E   FutureWarning: Setting an item of incompatible dtype is deprecated and will raise in a future error of pandas. Value '' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
```

I'm not a big `panda` or `metadata` user, not sure how best to solve it.

Appeared in https://github.com/mne-tools/mne-python/pull/12345

--- a/mne/epochs.py
+++ b/mne/epochs.py
@@ -2977,11 +2977,11 @@ def make_metadata(
         *last_cols,
     ]
 
-    data = np.empty((len(events_df), len(columns)))
+    data = np.empty((len(events_df), len(columns)), float)
     metadata = pd.DataFrame(data=data, columns=columns, index=events_df.index)
 
     # Event names
-    metadata.iloc[:, 0] = ""
+    metadata["event_name"] = ""
 
     # Event times
     start_idx = 1
@@ -2990,7 +2990,7 @@ def make_metadata(
 
     # keep_first and keep_last names
     start_idx = stop_idx
-    metadata.iloc[:, start_idx:] = None
+    metadata[columns[start_idx:]] = ""
 
     # We're all set, let's iterate over all events and fill in in the
     # respective cells in the metadata. We will subset this to include only

